packages/cheqd/src/CheqdApi.ts

```diff
@@ -0,0 +1,18 @@
+import { AgentContext } from '@credo-ts/core'
+import { injectable } from 'tsyringe'
+
+import { CheqdCreateResourceOptions, CheqdDidRegistrar } from './dids'
+
+@injectable()
+export class CheqdApi {
+  private agentContext: AgentContext
+
+  public constructor(agentContext: AgentContext) {
+    this.agentContext = agentContext
+  }
+
+  public async createResource(did: string, options: CheqdCreateResourceOptions) {
```

Comment:
In the underlying method, all fields except data are optional. Is this correct? Do we need to provide some simplified API top level? OR is it correct that all fields are optiona? (so just providing data should succeed)

Reply:
No @TimoGlastra, ideally the agent should provide the below attributes

- name
- resourceType
- data
-  id

They are defaulted to an empty string and will be rejected from the ledger side. Updated `CheqdCreateResourceOptions` with a stricter schema